baseplate/experiments/__init__.py

```diff
@@ -77,6 +78,16 @@ def _get_config(self, name):
             logger.warning("Could not load experiment config: %s", str(exc))
             return
 
+    def _get_experiment(self, name):
+        if name not in self._experiment_cache:
```

Comment:
The client should be re-fetching the experiment object in each new span context (i.e. each item fetched from the queue) since the user / data context has changed so I think maybe it's fine to keep the caching?